dolphinscheduler-registry/dolphinscheduler-registry-api/src/main/java/org/apache/dolphinscheduler/registry/api/RegistryClient.java

```diff
@@ -210,7 +210,7 @@ public boolean isMasterPath(String path) {
     }
 
     public boolean isWorkerPath(String path) {
-        return path != null && path.startsWith(RegistryNodeType.WORKER.getRegistryPath());
+        return path != null && path.startsWith(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS) && !path.equals(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS);
```

Comment:
```suggestion
        return path != null && path.startsWith(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS)
                && !path.equals(Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS);
```


@lxorc Do you have time to work on this PR? Maybe I can help with that. We also encountered this issue, and I have tested this PR in our environment. It is feasible.